src/main/java/duke/exception/DukeException.java

```diff
@@ -0,0 +1,22 @@
+package duke.exception;
+
+import duke.ui.Ui;
+
+/** Duke exception class to handle exceptions in the Duke program. */
+public class DukeException extends Exception {
+
+    /**
+     * Constructs a Duke Exception.
+     *
+     * @oaram message Message to be shown to user upon exception handling.
+     */
+    public DukeException(String message) {
+        super(message);
+    }
+
+    @Override
+    public String toString() {
+        return Ui.wrap(String.format("%s", getMessage()));
```

Comment:
Nice to see the use of String.format here! Although personally I would just cut straight to use `getMessage()` as the total String isn't very long